Java/core/src/main/java/com/amazon/randomcutforest/store/PointStore.java

```diff
@@ -179,6 +152,51 @@ public int decrementRefCount(int index) {
      */
     abstract void copyPoint(double[] point, int src, int location, int length);
 
+    /**
+     * takes an index from the index manager and rezises if necessary also adjusts
+     * refCount size to have increment/decrement be seamless
+     *
+     * @return an index from the index manager
+     */
+    int takeIndex() {
+        if (indexManager.freeIndexPointer == INFEASIBLE_POINTSTORE_INDEX) {
+            if (indexManager.getCapacity() < capacity) {
+                int oldCapacity = indexManager.getCapacity();
+                int newCapacity = Math.min(capacity, 2 * oldCapacity);
+                indexManager = new IndexManager(newCapacity);
+                for (int i = 0; i < oldCapacity; i++) {
+                    indexManager.occupied.set(i);
+                }
+                indexManager.freeIndexPointer = newCapacity - oldCapacity - 1;
```

Comment:
I think what we really want here is a copy constructor:
```
public IndexManager(IndexManager manager, int newCapacity) {
  this(newCapacity);
  occupied.or(manager.occupied);
  freeIndexPointer = newCapacity - manager.capacity;
}
```

Reply:
Done